### PR TITLE
SCION DNS server, to allow discovery of service instances.

### PR DESCRIPTION
**Pull request by kormat**
Based on work by @pszalach and @LorenzoBaesso.

DNS server polls ZK to find service instances, and exposes this
information over DNS for remote hosts to query.

Status:
- The DNS server works
- lib/dns.py is not finished yet
- edge_router needs updating to query DNS
- Unit tests not written yet

@pszalach

### CODE REVIEW

File: infrastructure/dns_server.py [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+"""
+# Stdlib
+import binascii
+import datetime
+import logging
+import os
+import sys
+import threading
+import time
+from ipaddress import (ip_address)
```
Comment (pszal) [inline infrastructure/dns_server.py]: parentheses ?

File: lib/dns.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+====================================
+"""
+# Stdlib
+import socket
+
+# External packages
+from dnslib import DNSRecord, QTYPE
+
+
+def get_hosts_by_srvname(service_name, dns_ip, dns_port=53, timeout=2):
```
Comment (pszal) [inline lib/dns.py]: dns_port=SCION_DNS_PORT ?

File: infrastructure/dns_server.py (hunk 1 above)
Comment (kormat) [inline infrastructure/dns_server.py]: Fixed

File: lib/dns.py (hunk 2 above)
Comment (kormat) [inline lib/dns.py]: Fixed, and added a comment saying this function isn't finished/tested/anything.

**Pull request merged**